src/libraries/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -12,9 +12,11 @@ internal static partial class ThrowHelper
         // If the exception source is this value, the serializer will re-throw as JsonException.
         public const string ExceptionSourceValueToRethrowAsJsonException = "System.Text.Json.Rethrowable";
 
-        public static ArgumentOutOfRangeException GetArgumentOutOfRangeException_MaxDepthMustBePositive(string parameterName)
+        [DoesNotReturn]
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
Generally adding NoInlining like this to these throw helpers is a deoptimization.  The JIT won't examine the body of a method marked as NoInlining, which means it won't see that it'll always throw and thus will have to emit the code for the call site under the expectation that the method call may return.  In contrast, without the attribute, it will look at the implementation and won't inline a method that always throws.